### ISSUE TITLE
JSON decoding issues starting with version 1.5.5 (Java::JavaLang::ArrayIndexOutOfBoundsException)

### ISSUE THREAD

**Issue by breml** [elastic/logstash#4316]
We suffered from ArrayIndexOutOfBounds exceptions while decoding JSON with logstash, starting from version 1.5.5.

My colleague described the problem in: https://discuss.elastic.co/t/json-en-decoding-issues-starting-with-version-1-5-5/36543

Now we could tackle down this issue. The problem is related to https://github.com/FasterXML/jackson-core/issues/216, which got fixed with jackson-* version 2.6.2.

I filed an issue in jrjackson to update the dependecies: https://github.com/guyboertje/jrjackson/issues/48

**Solution for logstash:**

As soon as jrjackson is updated, logstash should update to the latest version of jrjackson to fix this issue.

**Steps to reproduce:**

1. logstash config:

```
input {
  file {
    codec => json
    path => "/tmp/json-parse-problem.input"
  }
}
output {
  stdout {}
}
```
2. start logstash with the above config

3. provide some input

```
for i in `seq 1 120`; do echo '{"message":"'${i}'", "test'${i}'":0}'; done >> /tmp/json-parse-problem.input
```

The first message to throw an exception is number 114.

**Comment by jordansissel** [comment]
@guyboertje thoughts?

**Comment by guyboertje** [comment]
I will be creating a new version of JrJackson today to fix the Jackson bug.

**Comment by guyboertje** [comment]
@breml - released JrJackson today. v 0.3.8 in rubygems.  Are you able to do a `plugin update --no-verify` and get the new JrJackson version?

**Comment by breml** [comment]
@guyboertje I will check tomorrow

**Comment by breml** [comment]
@guyboertje we have checked with version 0.3.8 of jrjackson, and it is working fine now.

**Comment by guyboertje** [comment]
@breml - thank you for that.

**Comment by breml** [comment]
Upstream is fixed. Dependency in Logstash should be updated.

**Comment by suyograo** [comment]
This will be fixed in 2.1.2 and v2.2. We don't intend to release any more patch fixes on top of 1.5 unless its a security issue.

**Comment by suyograo** [comment]
@breml any reason you could not update to 2.x?

**Comment by breml** [comment]
Fix in v2.1.2 is fine for me.
